### PR TITLE
[DBZ-5014] fix infinity for date

### PR DESCRIPTION
**Pull request by cmartinez-enve**

### CODE REVIEW

**Comment by Naros** [comment]
Hi @cmartinez-enve, could you please adjust the commit message with DBZ-5014 to pass the commit format check?
Thanks!

**Comment by cmartinez-enve** [comment]
Yes i'm going to clean up the commits because I pushed a lot of garbage. And the intrgration test are not working

File: debezium-core/src/main/java/io/debezium/time/Date.java [hunk 1]
```diff
@@ -35,7 +35,7 @@ public class Date {
      * @return the schema builder
      */
     public static SchemaBuilder builder() {
-        return SchemaBuilder.int32()
+        return SchemaBuilder.string()
```
[the remark this answers is not in the document]
Comment (jpechane) [reply debezium-core/src/main/java/io/debezium/time/Date.java:38]: @cmartinez-enve Have you seen my comment, please?

[the remark this answers is not in the document]
Comment (cmartinez-enve) [reply debezium-core/src/main/java/io/debezium/time/Date.java:38]: yes, i'll push new commits in a moment, I have some problems with a unit test. I'm testing a delete scenario, when I validate if is a valid delete operation debezium (marks as a READ operation), maybe you can help me with this, thank you

**Comment by cmartinez-enve** [comment]
@jpechane This is the new version, without change the datatype of date, the problem is in my test case on delete.

in the file debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/RecordsStreamProducerIT.java line 3080, the debezium connector generates this error

`RecordsStreamProducerIT.shouldReceiveDeletesWithInfinityDate:233->assertDelete:3044->assertRecordDeleted:3073->assertRecordDeleted:3080 expected:<'[d]'> but was:<'[r]'>`

File: debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/RecordsStreamProducerIT.java [hunk 2]
```diff
@@ -3033,6 +3066,26 @@ private SourceRecord assertRecordInserted(SourceRecord insertedRecord, String ex
...
+        SourceRecord insertedRecord = consumer.remove();
+
+        return assertRecordDeleted(insertedRecord, expectedTopicName, pkColumn, pk);
+    }
+
+    private SourceRecord assertRecordDeleted(SourceRecord deletedRecord, String expectedTopicName, String pkColumn, Integer pk) throws InterruptedException {
+        assertEquals(topicName(expectedTopicName), deletedRecord.topic());
+
+        if (pk != null) {
+            VerifyRecord.isValidDelete(deletedRecord, pkColumn, pk);
```
Comment (cmartinez-enve) [inline debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/RecordsStreamProducerIT.java:3080]:
```
org.junit.ComparisonFailure: expected:<'[d]'> but was:<'[r]'>
	at io.debezium.connector.postgresql.RecordsStreamProducerIT.assertRecordDeleted(RecordsStreamProducerIT.java:3080)
	at io.debezium.connector.postgresql.RecordsStreamProducerIT.assertRecordDeleted(RecordsStreamProducerIT.java:3073)
	at io.debezium.connector.postgresql.RecordsStreamProducerIT.assertDelete(RecordsStreamProducerIT.java:3044)
	at io.debezium.connector.postgresql.RecordsStreamProducerIT.shouldReceiveDeletesWithInfinityDate(RecordsStreamProducerIT.java:233)
```

File: debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/RecordsStreamProducerIT.java [hunk 3]
```diff
@@ -3033,6 +3065,26 @@ private SourceRecord assertRecordInserted(SourceRecord insertedRecord, String ex
...
+        assertFalse("records not generated", consumer.isEmpty());
+        SourceRecord insertedRecord = consumer.remove();
+
+        return assertRecordDeleted(insertedRecord, expectedTopicName, pkColumn, pk);
+    }
+
+    private SourceRecord assertRecordDeleted(SourceRecord deletedRecord, String expectedTopicName, String pkColumn, Integer pk) throws InterruptedException {
+        assertEquals(topicName(expectedTopicName), deletedRecord.topic());
+
+        // if (pk != null) {
```
Comment (jpechane) [inline debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/RecordsStreamProducerIT.java:3078]: Why is this commented out? Is this related to the error you are mentioning?

**Comment by jpechane** [comment]
@cmartinez-enve I am sorry but only now I noticed the PR is opened against `1.9` branch. Could you please open it against `main`? We apply changes to `main` first nd then we backport it to stable branch. Thanks for understanding!

File: debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/RecordsStreamProducerIT.java (hunk 3 above)
Comment (cmartinez-enve) [reply debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/RecordsStreamProducerIT.java]: Yes, I uncommented this code

**Comment by jpechane** [comment]
@cmartinez-enve I believe the tests are fixed - let's wait for CI test results.

**Comment by cmartinez-enve** [comment]
It's working now, Thank you

**Comment by jpechane** [comment]
@cmartinez-enve Applied, thanks!

**Comment by cmartinez-enve** [comment]
@jpechane Do you have any idea when the next release will be done?

**Comment by jpechane** [comment]
@cmartinez-enve I suppose in two weeks.
